gensrc/thrift/FrontendService.thrift

```diff
@@ -752,18 +753,33 @@ struct TSchemaMeta {
 
 struct TIndexMeta {
     1: optional i64 index_id
-    2: optional string index_state
-    3: optional i64 row_count
-    4: optional i64 rollup_index_id
-    5: optional i64 rollup_finished_version
-    6: optional TSchemaMeta schema_meta
-    7: optional list<TTabletMeta> tablets
+    2: optional i64 partition_id
+    3: optional string index_state
+    4: optional i64 row_count
+    5: optional i64 rollup_index_id
+    6: optional i64 rollup_finished_version
+    7: optional TSchemaMeta schema_meta
+    8: optional list<TTabletMeta> tablets
+}
+
+struct TDataProperty {
+    1: Types.TStorageMedium storage_medium
+    2: i64 cold_time
+}
+
+struct TRangePartitionDesc {
+    1: optional i64 partition_id
+    2: optional binary start_key
+    3: optional binary end_key
 }
 
 struct TPartitionInfo {
     1: optional string type
     2: optional map<i64, i16> replica_num_map
     3: optional map<i64, bool> in_memory_map
+    4: optional list<TColumnMeta> columns
+    5: optional map<i64, TRangePartitionDesc> partition_desc
+    6: optional map<i64, TDataProperty> data_property
```

Comment:
StarRocks supprot multiple types of partition.
I think it is a good way to create a struct for each partition type. such as a TRangePartitionDesc, TSinglePartitionDesc. When `list partition` is supported, TListPartitionDesc can be supported.